### DISCUSSION TITLE
Portal Administration

### DISCUSSION THREAD

**Discussion by Noggling** [equinor/fusion-project-portal#404] [Ideas]
The application will appear under app category Portal utilities alongside Service message application and the context explorer

As our project portal continues to expand with more contexts, the demand for an administration tool has become apparent. 
This tool primarily serves developers and portal administrators, empowering them to effectively manage the project portal.

To ensure secure and controlled access, this administration tool should be developed as a Fusion application and made accessible exclusively to users possessing the Fusion.ProjectPortal.Admin role.

> [!IMPORTANT] 
> In order to maintain a seamless user experience, the application will not be visible in the menu unless the user has the required role. This access control will be enforced at the backend API level to prevent unauthorized entry.

This approach embraces role-based service applications, where access is contingent upon specific roles. The administration tool will be categorized under "Portal Utilities," alongside other applications such as the Service Message application and the Context Explorer.

## What is needed 

The Portal Administration Tool must fulfill the following acceptance criteria:

1. **Administrative Work Surface:** The tool should enable administrative users to effectively work with and view project facilities and contracts.

2. **Context Onboarding:** Administrators should have the capability to onboard new contexts, making it a seamless process.

3. **Application Onboarding:** Users should be able to onboard new applications, simplifying the application integration process.

4. **Connect Applications and Contexts:** The tool should facilitate the connection of applications with specific contexts, ensuring that they operate harmoniously within the portal environment.

###  Work Surface

As a Portal Administrator, I want to have the ability to create, edit, and delete 'Work Surfaces.' 

A 'Work Surface' represents a user's perspective within the project portal. It empowers the administrator to define which contexts are accessible within the designated 'Work Surface.' 

A 'Work Surface' comprises the following attributes:

- Name: A descriptive title for the Work Surface.
- Short Name: A concise identifier for quick reference.
- Description: A textual explanation of the Work Surface's purpose.
- Icon: A visual representation to easily identify the Work Surface.
- List of Available Context Types: A selection of context types that can be accessed within this Work Surface.

The administrator should be able to edit these attributes and assign context types as needed to tailor the Work Surface to specific user requirements.

The administrator should also have the capability to arrange the order of work surfaces and designate the default work surface.

### Application Groups

As an Application and Portal Administrator, I need the ability to manage application groups. Application groups are used for organizing applications within the menu.

An application group is defined by the following attributes:

- Name: A descriptive label for the application group.
- ID: A unique identifier.
- Color: A visual cue for distinguishing the group.
- Description: An optional explanation of the group's purpose.

In my role as an administrator, I should be able to modify and update these attributes as necessary.

Additionally, when an application group is created, I should be able to onboard applications using the application group and an application key. This will streamline the process of associating applications with specific groups for efficient menu organization.

### Application Onboarding

As an Application and Portal Administrator, I require the capability to onboard applications to the project portal. In this process, I need to associate applications with specific application groups. This can be achieved by selecting an application from the list of available applications and assigning it to the appropriate application group. This functionality will enable me to effectively manage and organize applications within the portal, ensuring a well-structured and user-friendly menu system.

> [!WARNING] 
> The administrator should have the ability to specify whether the application is a legacy Fusion application that necessitates a custom application loader.

### Application Context Link

As an Application and Portal Administrator, I require the capability to associate an application with one or multiple contexts within the project portal. Additionally, I should have the flexibility to designate the application as a global resource accessible across all contexts when needed. This functionality ensures that I can effectively control the application's availability and relevance, tailoring it to specific context while maintaining the option for broader access when required.

> [!IMPORTANT] 
> link to the already created milestone
> [Portal Administration](https://github.com/equinor/fusion-project-portal/milestone/9)  

![GitHub milestone details](https://img.shields.io/github/milestones/progress-percent/equinor/fusion-project-portal/9)
![Static Badge](https://img.shields.io/badge/Needs_grooming-5555ee)

**Comment by Noggling** [comment]
> [!IMPORTANT] 
>For additional backend information regarding portal administration, please refer to the following resources.
https://equinor.github.io/fusion-project-portal-internal/docs/developer/admin/administration

**Comment by AreWiv** [comment]
Reviewed and understood.
